### ISSUE TITLE
Mono: OSX exports build errors with System.Net.Http namespace and possibly System.Net.ServicePointManager initialization

### ISSUE THREAD

**Issue by newnoiseworks** [godotengine/godot#27830]
**Godot version:**
3.1 mono official

**OS/device including version:**
Windows 10 && OSX Mojave 10.14.3 (built on both machines)

**Issue description:**
My game uses Nakama, a third party open source backend, and their .NET client, which relies on the core `System.Net.Http` library. Everything works fine in the editor in Windows 10 && OSX, and in exported builds for Windows.

However, when attempting to export _to_ OSX _from_ either machine, I get the same error in the exported build when attempting to login, where it appears the `System.Net.Http` library hasn't been bundled in the exported build (or something, I'm not sure, a bit new to Mono's package management). Below is the error reproduced using just the core `System.Net.Http` API (linked repo + zip below as well).

```csharp
error! System.NullReferenceException: Object reference not set to an instance of an object.
  at System.Net.ServicePointManager.FindServicePoint (System.Uri address, System.Net.IWebProxy proxy) [0x0010d] in <9d34f015e4174611857eb4e1bfe180ff>:0 
  at System.Net.HttpWebRequest.GetServicePoint () [0x00021] in <9d34f015e4174611857eb4e1bfe180ff>:0 
  at System.Net.HttpWebRequest.set_Proxy (System.Net.IWebProxy value) [0x00014] in <9d34f015e4174611857eb4e1bfe180ff>:0 
  at System.Net.Http.HttpClientHandler.CreateWebRequest (System.Net.Http.HttpRequestMessage request) [0x00108] in <e27dc36b32de428a884dd0a2efb5e298>:0 
  at System.Net.Http.HttpClientHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x0003e] in <e27dc36b32de428a884dd0a2efb5e298>:0 
  at System.Net.Http.HttpClient.SendAsyncWorker (System.Net.Http.HttpRequestMessage request, System.Net.Http.HttpCompletionOption completionOption, System.Threading.CancellationToken cancellationToken) [0x000ca] in <e27dc36b32de428a884dd0a2efb5e298>:0 
  at Node2D._Ready () [0x0009a] in <1db139ff92d04c979876ce0417769785>:0 
```

**Steps to reproduce:**
Either use the examples below, or setup a project with Nakama (ha), or setup a new project in Godot 3.1 mono and make a `System.Http.HttpClient#SendAsync` call - I was posting `GET` to `http://google.com` and getting back HTML just fine in the editor.

Now export an OSX build using the latest export template; it should return the above error (my project below tracks this in a `TextEdit` field on screen.

**Minimal reproduction project:**
Grab a .zip from the release page and open it in Godot 3.1:
https://github.com/newnoiseworks/GodotMonoNetHttpOSX/releases/tag/0.0.1

Or just check the code out real quick:
https://github.com/newnoiseworks/GodotMonoNetHttpOSX

**Comment by newnoiseworks** [comment]
update: I talked to the dev @ nakama about the `System.Net.Http` issue and they think it has something to do with `System.Net.ServicePointManager` not being initialized correctly, per the repo owner:

> Yep the `ServicePointManager` is used to set network proxy settings with HTTP requests in the .NET runtime. Looks like however the Godot team have packaged the .NET runtime it doesn't initialize that class correctly.

**Comment by cart** [comment]
I have reproduced this in my game High Hat. This is a blocker for online matchmaking

**Comment by mastern2k3** [comment]
Bump, I'm having this problem too.

**Comment by akien-mga** [comment]
Is this still reproducible in latest 3.2 builds?

**Comment by cart** [comment]
Unfortunately I get this whenever i export to macos

```
OpenGL ES 2.0 Renderer: NVIDIA GeForce GT 750M OpenGL Engine
 
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
ERROR: _init_godot_api_hashes: Mono: Core API hash mismatch.
   At: modules/mono/mono_gd/gd_mono.cpp:465.
Mono: Logfile is: /Users/carter/Library/Application Support/Godot/app_userdata/HighHat/mono/mono_logs/2020_01_16 21.09.44 (510).txt
ERROR: _load_api_assemblies: The assembly 'GodotSharp' is out of sync.
   At: modules/mono/mono_gd/gd_mono.cpp:926.
ERROR: _load_api_assemblies: FATAL: Method/Function Failed.
   At: modules/mono/mono_gd/gd_mono.cpp:937.

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGILL while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x1003a7d38 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : mono_dump_native_crash_info
	0x10039b5a5 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : mono_handle_native_crash
	0x1003aef50 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : mono_sigill_signal_handler
	0x7fff5af54b5d - /usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x0 - Unknown
	0x1009cdf7e - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : _ZN6GDMono26initialize_load_assembliesEv
	0x100954d91 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : _ZN14CSharpLanguage4initEv
	0x101f50f59 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : _ZN12ScriptServer14init_languagesEv
	0x10062a4e2 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : _ZN4Main6setup2Ey
	0x100624b9f - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : _ZN4Main5setupEPKciPPcb
	0x10060b819 - /Users/carter/Library/Application Support/itch/apps/high-hat/HighHat.app/Contents/MacOS/./HighHat : main
	0x7fff5ad693d5 - /usr/lib/system/libdyld.dylib : start
	0x1 - Unknown

=================================================================
	Telemetry Dumper:
=================================================================
Pkilling 0x70000e817000 from 0x1128845c0
* Assertion: should not be reached at threads.c:6254
Entering thread summarizer pause from 0x1128845c0
Finished thread summarizer pause from 0x1128845c0.

Waiting for dumping threads to resume
```

**Comment by cart** [comment]
Thats using 3.2 beta 6

**Comment by newnoiseworks** [comment]
@akien-mga I just gave the demo project I linked to in the OP a try w/ Godot 3.2 RC1 and it worked!

@cart don't want to read too deeply into the stack trace you have but it seems your error may be related to another area of your project / Godot?

Anyway I'm gonna do a bit more testing -- so far I've built a version of the Godot Export to OSX *from* Windows. Will try building to OSX etc, but, fingers crossed things'll work well
